router/core/server.go

```diff
@@ -167,6 +170,14 @@ func (s *server) buildMux(ctx context.Context,
 		planCache = NewNoopExecutionPlanCache()
 	}
 
+	// Close the plan cache when the context is done.
+	// This is happening after the server is shutdown.
+	// This will stop all goroutines and free up resources
+	go func() {
+		<-ctx.Done()
+		planCache.Close()
```

Comment:
We should also clean up and close the ePoll instance here, right?
What about WebSockets in general? If we update the config, it's possible that the origins can no longer serve the old contract. Should we terminate and cleanup all WebSockets, independent of ePoll or normal connection handling.

Reply:
Epoll is closed on context cancellation (introduced in this PR) and cleaned up in a defer.

Reply:
In the future, we should wait until all resources has been shutdown before we restart the server. For now this is done asynchronously but sufficient.